gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -317,26 +312,27 @@ public void orchestrate(Spec spec, Properties jobProps, long triggerTimestampMil
       }
 
       // If multi-active scheduler is enabled do not pass onto DagManager, otherwise scheduler forwards it directly
-      if (this.isMultiActiveSchedulerEnabled) {
+      if (schedulerLeaseAlgoHandler.isPresent()) {
+        // If triggerTimestampMillis is 0, then it was not set by the job trigger handler, and we cannot handle this event
+        if (triggerTimestampMillis == 0L) {
+          _log.warn("Skipping execution of spec: {} because missing trigger timestamp in job properties",
+              jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY));
+          flowMetadata.put(TimingEvent.METADATA_MESSAGE, "Flow orchestration skipped because no trigger timestamp "
+              + "associated with flow action.");
+          if (this.eventSubmitter.isPresent()) {
+            new TimingEvent(this.eventSubmitter.get(), TimingEvent.FlowTimings.FLOW_FAILED).stop(flowMetadata);
```

Comment:
tip: `ifPresent()`

(note: this works easily and naturally... unless checked exceptions)

Reply:
This format is used in many other places in the `Orchestrator` so I will leave as is